### PR TITLE
Desmod System-based IK Rewrite

### PR DESCRIPTION
**Pull request by PY44N**

### CODE REVIEW

File: src/main/kotlin/frc/robot/RobotStateMachine.kt [hunk 1]
```diff
@@ -34,11 +34,11 @@ enum class IntakeState(val innerPercent: Double, val outerPercent: Double) {
 }
 
 //this represents the DESIRED trunk statautoshooe
-enum class TrunkPosition(val angle: Double, val position: Double) {
+enum class TrunkPose(val angle: Double, val position: Double) {
```
Comment (PY44N) [inline src/main/kotlin/frc/robot/RobotStateMachine.kt:37]: Why did you rename this? The new name isn't more descriptive, and it just clogs up the changes

File: src/main/kotlin/frc/robot/subsystems/trunk/TrunkIOReal.kt [hunk 2]
```diff
@@ -16,15 +16,23 @@ class TrunkIOReal : TrunkIO {
...
-    private var topPositionLimit = TrunkConstants.TOP_BREAK_BEAM_POSITION
-    private var bottomRotationLimit = TrunkConstants.SAFE_TRAVEL_ANGLE
-    private var topRotationLimit = TrunkConstants.MAX_ANGLE
+    override var positionBrake = true
+        set(enabled) {
+            if (positionBrake != enabled)
+                elevatorMotor.setIdleMode(if (enabled) CANSparkBase.IdleMode.kBrake else CANSparkBase.IdleMode.kCoast)
+            field = enabled
+        }
+    override var rotationBrake = true
```
Comment (PY44N) [inline src/main/kotlin/frc/robot/subsystems/trunk/TrunkIOReal.kt:28]: This seems complicated for the small amount of CAN bus bandwidth its saving

File: src/main/kotlin/frc/robot/RobotStateMachine.kt (hunk 1 above)
Comment (dezash123) [reply src/main/kotlin/frc/robot/RobotStateMachine.kt:37]: calling it position is confusing because it contains more than position
